src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,568 @@
+
+    //Modifications
+    registerStringList_("fixed_modifications", "<mods>", ListUtils::create<String>(""), "Fixed modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Carbamidomethyl (C)' or 'Oxidation (M)'", false, false);
+    vector<String> all_mods;
+    ModificationsDB::getInstance()->getAllSearchModifications(all_mods);
+    setValidStrings_("fixed_modifications", all_mods);
+    registerStringList_("variable_modifications", "<mods>", ListUtils::create<String>(""), "Variable modifications, specified using UniMod (www.unimod.org) terms, e.g. 'Carbamidomethyl (C)' or 'Oxidation (M)'", false, false);
+    setValidStrings_("variable_modifications", all_mods);
+    registerIntOption_("max_variable_mods_in_peptide", "<num>", 5, "", false, true);
+    
+  }
+
+  vector<ResidueModification> getModifications_(StringList modNames)
+  {
+   vector<ResidueModification> modifications;
+
+    // iterate over modification names and add to vector
+    for (StringList::iterator mod_it = modNames.begin(); mod_it != modNames.end(); ++mod_it)
+    {
+      String modification(*mod_it);
+      modifications.push_back(ModificationsDB::getInstance()->getModification(modification));
+    }
+
+    return modifications;
+  }
+
+  void removeTempDir_(const String& tmp_dir)
+  {
+    if (tmp_dir.empty()) return; // no temp. dir. created
+
+    if (debug_level_ >= 2)
+    {
+      writeDebug_("Keeping temporary files in directory '" + tmp_dir + "'. Set debug level to 1 or lower to remove them.", 2);
+    }
+    else
+    {
+      if (debug_level_ == 1) writeDebug_("Deleting temporary directory '" + tmp_dir + "'. Set debug level to 2 or higher to keep it.", 1);
+      File::removeDirRecursively(tmp_dir);
+    }
+  }
+
+  void createParamFile_(ostream& os)
+  {
+    os << "# comet_version 2016.01 rev. 2\n";               //required as first line in the param file
+    os << "# Comet MS/MS search engine parameters file.\n";
+    os << "# Everything following the '#' symbol is treated as a comment.\n";
+    os << "database_name = " << getStringOption_("database") << "\n";
+    os << "decoy_search = " << 0 << "\n"; // 0=no (default), 1=concatenated search, 2=separate search
+    os << "num_threads = " << getIntOption_("threads") << "\n";  // 0=poll CPU to set num threads; else specify num threads directly (max 64)
+
+    // masses
+    os << "peptide_mass_tolerance = " << getDoubleOption_("precursor_mass_tolerance") << "\n";
+    os << "peptide_mass_units = " << getIntOption_("precursor_error_units") << "\n";                  // 0=amu, 1=mmu, 2=ppm
+    os << "mass_type_parent = " << 1 << "\n";                    // 0=average masses, 1=monoisotopic masses
+    os << "mass_type_fragment = " << 1 << "\n";                  // 0=average masses, 1=monoisotopic masses
+    os << "precursor_tolerance_type = " << 1 << "\n";            // 0=MH+ (default), 1=precursor m/z; only valid for amu/mmu tolerances
```

Comment:
hmm not sure if this might be a problem if ppm tolerances are used. please check